### PR TITLE
Use unsigned types to avoid signed integer overflow in address comput…

### PR DESCRIPTION
**Pull request by dougkwan**
This fixes a problem in which signed integer types are used in address computation.  Signed integer overflow is an undefined behavior in C. The problem was discovered by fuzzing XED.

### CODE REVIEW

File: src/dec/xed-agen.c [hunk 1]
```diff
@@ -38,7 +38,7 @@ xed_error_enum_t xed_agen(xed_decoded_inst_t* xedd,
     xed_uint64_t index_value = 0;
     xed_uint64_t segment_base = 0;
     xed_uint64_t scale  = 0;
-    xed_int64_t displacement = 0;
+    xed_uint64_t displacement = 0;
```
Comment (marjevan) [inline src/dec/xed-agen.c:41]: Memory displacement is a signed value. Why did you change it here?

Intel SDM:
image

Comment (dougkwan) [reply src/dec/xed-agen.c:41]: The signedness of displacement here may not matter as the value it holds will be promoted to an unsigned type when adding to an unsigned value like base.  However, it could be a potential bug if a narrower signed type is assigned to it.  Luckily it is assigned only once with and xed_decoded_inst_get_memory_displacement() returns init64_t so no signed extension is involved.  I can change this commit so that all displacement variable are signed type.

Comment (dougkwan) [reply src/dec/xed-agen.c]: I have updated the commit to not change all displacement variables to unsigned types.

**Comment by marjevan** [comment]
I'll integrate a solution that will be available with the next XED external release.
Thanks!
